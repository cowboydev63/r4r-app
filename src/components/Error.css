.r4r-error.r4r-DEFAULT {
    height: 100%;
    min-height: 300px;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    color: white;
    background: linear-gradient(304deg, #ec0000, #6407e0);
    background-size: 400% 400%;
    
    animation: AnimationName 5s ease infinite;
}

.r4r-error.r4r-DEFAULT h1, .r4r-error.r4r-DEFAULT h3 {
    color: white;
}

.r4r-error.r4r-DEFAULT h3 {
    margin: 15px 0;
}

@-webkit-keyframes AnimationName {
    0%{background-position:0% 5%}
    50%{background-position:100% 96%}
    100%{background-position:0% 5%}
}
@-moz-keyframes AnimationName {
    0%{background-position:0% 5%}
    50%{background-position:100% 96%}
    100%{background-position:0% 5%}
}
@keyframes AnimationName { 
    0%{background-position:0% 5%}
    50%{background-position:100% 96%}
    100%{background-position:0% 5%}
}