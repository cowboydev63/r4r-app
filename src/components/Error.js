import React from 'react';
import PropTypes from 'prop-types';

class Error extends React.PureComponent {
    static propTypes = {
        message: PropTypes.string,
    }

    static defaultProps = {
        message: 'Unknown Error: please try refreshing',
    }

    render(){
        return(
            <div>
                <h1>Error</h1>
                <p>{ this.props.message }</p>
            </div>
        )
    }
}

export default Error;