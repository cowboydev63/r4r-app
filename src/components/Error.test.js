import React from 'react';
import { shallow } from 'enzyme';
import Error from './Error';

describe('Error Component', () => {
  it('renders without crashing', () => {
    shallow(<Error />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Error />);
    expect(wrapper).toMatchSnapshot();
  })
})