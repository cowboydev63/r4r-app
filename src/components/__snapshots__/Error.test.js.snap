// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component renders correctly 1`] = `
<Theme
  className="r4r-error"
  element="div"
>
  <h1>
    Error:
  </h1>
  <h3>
    Unknown Error: please try refreshing
  </h3>
  <p>
    Please try refreshing the page
  </p>
</Theme>
`;
